Sources/Datadog/Core/Feature.swift

```diff
@@ -32,11 +32,17 @@ internal struct FeaturesCommonDependencies {
 }
 
 internal struct FeatureStorage {
-    /// Writes data to files.
+    /// Writes data to files. This `Writer` takes current value of the `TrackingConsent` into consideration
+    /// to decided if the data should be written to authorized or unauthorized folder.
     let writer: Writer
-    /// Reads data from files.
+    /// Reads data from files in authorized folder.
     let reader: Reader
 
+    /// An arbitrary `Writer` which always writes data to authorized folder.
```

Comment:
I tried to find a way that says _"it doesn't make any consideration of `TrackingConsent` when writing data and writes it to authorised folder all the time"_. Agree 👌, the "arbitrary" may not be the perfect choice. Do you think of a better name? It's quite important piece in the architecture 🙂 as you point out in https://github.com/DataDog/dd-sdk-ios/pull/407#discussion_r574388312.

Reply:
I wasn't suggesting to change it as it's a valid meaning, just pointing the potential confusion (and as a note for future me when git-blaming/coming back to this PR).
I don't have a concise alternative but otherwise a verbose one could be `TrackingConsentAgnosticWriter`. But that's tool long already. I think that the current naming is sensible and we can keep it that way.